RecoLocalTracker/SiPixelRecHits/interface/PixelCPEGenericESProducer.h

```diff
@@ -11,8 +11,9 @@ class  PixelCPEGenericESProducer: public edm::ESProducer{
  public:
   PixelCPEGenericESProducer(const edm::ParameterSet & p);
   ~PixelCPEGenericESProducer() override; 
-  std::unique_ptr<PixelClusterParameterEstimator> produce(const TkPixelCPERecord &);
+  std::shared_ptr<PixelClusterParameterEstimator> produce(const TkPixelCPERecord &);
```

Comment:
This was changed recently from `shared_ptr` to `unique_ptr` in #22031. I believe it would be better to stay with `unique_ptr`.
@Dr15Jones

Reply:
unique_ptr is much better.

Reply:
Just to be clear: you want me to re-implement changes made by @gartung  in #22031, right?

Reply:
BTW, the change back to shared_ptr was not intentional.  #22031 came while this was in development.  I think I need to learn how to follow changes in pixel-related packages, since I could have started with #22031 in the first place :-/

Reply:
> Just to be clear: you want me to re-implement changes made by @gartung in #22031, right?

Yes, exactly (or otherwise undo your changes). 

> BTW, the change back to shared_ptr was not intentional. #22031 came while this was in development. 

That's what I thought happened. I''ve found that with git it is easiest to think in terms of diffs / commits, and when changing releases move the diffs / commits (e.g. with cherry-pick) rather than copying files.

> I think I need to learn how to follow changes in pixel-related packages, since I could have started with #22031 in the first place :-/

See http://cms-sw.github.io/faq.html#how-do-i-receive-notifications-about-pull-requests-for-a-given-package--subsystem
(you can even edit the in GitHub web UI so there is no need to clone the repo to yourself)